### PR TITLE
Grpc bazel build support

### PR DESCRIPTION
**Pull request by makdharma**
WORKSPACE, BUILD and other supporting files needed to get the google.monitoring.v3.metric_service building with bazel.

### CODE REVIEW

File: bazel/cc_grpc_library.bzl [hunk 1]
```diff
@@ -0,0 +1,61 @@
+"""Generates and compiles C++ grpc stubs from proto_library rules."""
```
Comment (ethanbao) [inline bazel/cc_grpc_library.bzl]: What are the difference between bzl extension in this PR vs the one already in grpc github repo (https://github.com/grpc/grpc/blob/master/grpc.bzl)? And why these grpc bzl files part of googleapis not grpc?

File: google/protobuf/duration.proto [hunk 2]
```diff
@@ -0,0 +1,104 @@
+// Protocol Buffers - Google's data interchange format
```
Comment (ethanbao) [inline google/protobuf/duration.proto]: Adding protobuf protos into googleapis will cause confusion, as the same protos are already in https://github.com/google/protobuf/tree/master/src/google/protobuf

File: bazel/cc_grpc_library.bzl (hunk 1 above)
Comment (makdharma) [inline bazel/cc_grpc_library.bzl]: Got rid of the custom copy.

File: google/protobuf/duration.proto (hunk 2 above)
Comment (makdharma) [inline google/protobuf/duration.proto]: Unfortunately grpc bazel rules don't work with well known protos yet. Added a TODO to get rid of local copy when grpc supports this capability.

Comment (ethanbao) [inline google/protobuf/duration.proto]: What are other options other than adding the protobuf protos here? Protos in this repo will be cloned/branched by developers, so we don't want to add temporary/incomplete protos here.

Comment (makdharma) [inline google/protobuf/duration.proto]: The only other option (until first class support is available) is to create a shell script that does the copying of native protobuf files in preparation of running ```bazel build```.

Comment (makdharma) [inline google/protobuf/duration.proto]: Never mind, we have a fix in the grpc tree. I'll modify this PR soon and remove local copy of google/protobuf.

Comment (ethanbao) [inline google/protobuf/duration.proto]: Thanks for the fix.

**Comment by makdharma** [comment]
@ethanbao PTAL

**Comment by ethanbao** [comment]
LGTM

**Comment by makdharma** [comment]
who needs to merge this PR?

**Comment by qiwzhang** [comment]
This is bad.  This breaks many other project bazel builds.  We should not have any BUILD files in any of sub folders.

Other projects, such as Cloud endpoints,  are using their own BUILD files based on their own WORKSPACE dependency to build these proto files.

Here is the Could endpoint bazel BUILD file.

https://github.com/istio/proxy/blob/master/contrib/endpoints/repositories.bzl#L225

You can have BUILD file at the top, but not inside the subfolders.   When Endpoints BUILD file trying to use proto files inside the folder,  it if already has a BUILD file,  bazel will fail.

Can we roll this back?

**Comment by makdharma** [comment]
No problem to roll it back. @ethanbao needs to do it?

**Comment by nicolov** [comment]
I agree, I think you should roll back this change and have a single top-level BUILD file. Including this in a `new_http_archive` makes bazel read the nested BUILD files, and fail.

**Pull request merged**